File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        
        HashMap<String, Boolean> phoneBookMap = new HashMap<>();
        
        for(String p: phone_book){
            phoneBookMap.put(p, true);
        }
        
        for(String k: phoneBookMap.keySet()){
            for(int i=1; i<k.length(); i++){
                String target = k.substring(0,i);
                // System.out.println(target);
                if(phoneBookMap.getOrDefault(target, false)){
                    return false;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    var answer = true;
    
    const map = new Map();
    
    for(let phone of phone_book){
        map.set(phone, true);
    }
    
    for(let phone of map.keys()){
        for(let i=1; i<phone.length; i++){
            if(map.has(phone.substring(0, i))){
                return false;
            }
        }
    }
    
    return answer;
}